### ISSUE TITLE
[Bug] Not all areas are being recolored

### ISSUE THREAD

**Issue by avi12** [gilday/dark-mode-sync-plugin#18]
![phpstorm64_2020-05-23_23-07-52](https://user-images.githubusercontent.com/6422804/82739686-3e80f180-9d4a-11ea-9f56-c9b82c24448e.png)

OS: Windows 10 Pro 1909 build 18363.720
IDE: PhpStorm 2020.1.1
Plugin version: 1.2.2

Note that the behavior doesn't always occur, but when it does - I can't seem to be able to get the whole IDE to be recolored.

**Comment by eldicn** [comment]
**I also encountered this problem.** 

This is my environment

```
OS: macOS 10.15.4
IDE: IntelliJ IDEA 2020.1.2 (Ultimate Edition)
Plugin: 1.2.2
```

done

**Comment by gilday** [comment]
I used to experience this as well, but this hasn't happened to me in a while so I assumed it was fixed. If it happens again, lmk

**Comment by avi12** [comment]
I will
For the moment, please update your plugin to the IntelliJ IDEA products version 2020.2

**Comment by BorzdeG** [comment]
@avi12 I used the latest version in IDEA 2020.2 - issue #22

**Comment by gilday** [comment]
This one stumps me. The inconsistency of this bug feels like a concurrency issue.

As far as I can tell, this plugin uses the same API that IntelliJ's `com.intellij.ide.actions.QuickChangeLookAndFeel` action uses

```java
QuickChangeLookAndFeel.switchLafAndUpdateUI(lafManager, newLaf, true);
```

The one difference is I set the `async` boolean parameter to `true`, because the background thread on which the plugin operates is not a UI thread, so by setting `async` to `true`, I see that IntelliJ will schedule the LAF update on the UI thread.

```java
    if (async) {
      //noinspection SSBasedInspection
      SwingUtilities.invokeLater(updater);
    }
```

I don't know why this doesn't work

**Comment by BorzdeG** [comment]
@gilday Open an issue with JetBrains?

**Comment by gilday** [comment]
I believe we might be experiencing https://youtrack.jetbrains.com/issue/IDEA-197105

I have also raised this issue in the JetBrains Platform Slack channel intellij-platform

**Comment by weisJ** [comment]
I have experienced the same issue with [Auto Dark Mode](https://github.com/weisJ/auto-dark-mode/issues/2). The implementation in IntelliJ to switch the theme actually uses the `Alarm` class to schedule the request. This seemed to resolve the issue for me (beside text color in the gradle tab not changing). [Here](https://github.com/weisJ/auto-dark-mode/blob/master/plugin/src/main/java/com/github/weisj/darkmode/AutoDarkMode.kt) is the relevant implementation.

**Comment by gilday** [comment]
I think `Alarm` is exactly what we need. Thanks @weisJ !

Released a fix in v1.2.5
